src/server_manager/web_app/app.ts

```diff
@@ -855,42 +858,82 @@ export class App {
-      await this.selectedServer.setAccessKeyDataLimit(limit);
+      await this.selectedServer.setDefaultDataLimit(limit);
       this.appRoot.showNotification(this.appRoot.localize('saved'));
-      serverView.accessKeyDataLimit = dataLimitToDisplayDataAmount(limit);
+      serverView.defaultDataLimitBytes = limit?.bytes;
+      serverView.isDefaultDataLimitEnabled = !!limit;
       this.refreshTransferStats(this.selectedServer, serverView);
       // Don't display the feature collection disclaimer anymore.
       serverView.showFeatureMetricsDisclaimer = false;
       window.localStorage.setItem('dataLimits-feature-collection-notification', 'true');
     } catch (error) {
-      console.error(`Failed to set access key data limit: ${error}`);
+      console.error(`Failed to set server default data limit: ${error}`);
       this.appRoot.showError(this.appRoot.localize('error-set-data-limit'));
-      serverView.accessKeyDataLimit = dataLimitToDisplayDataAmount(
-          previousLimit || await computeDefaultAccessKeyDataLimit(this.selectedServer));
-      serverView.isAccessKeyDataLimitEnabled = !!previousLimit;
+      const defaultLimit = previousLimit || await computeDefaultDataLimit(this.selectedServer);
+      serverView.defaultDataLimitBytes = defaultLimit?.bytes;
+      serverView.isDefaultDataLimitEnabled = !!previousLimit;
     }
   }
 
-  private async removeAccessKeyDataLimit() {
+  private async removeDefaultDataLimit() {
     const serverView = this.appRoot.getServerView(this.appRoot.selectedServerId);
     try {
-      await this.selectedServer.removeAccessKeyDataLimit();
+      await this.selectedServer.removeDefaultDataLimit();
       this.appRoot.showNotification(this.appRoot.localize('saved'));
       this.refreshTransferStats(this.selectedServer, serverView);
     } catch (error) {
-      console.error(`Failed to remove access key data limit: ${error}`);
+      console.error(`Failed to remove server default data limit: ${error}`);
       this.appRoot.showError(this.appRoot.localize('error-remove-data-limit'));
-      serverView.isAccessKeyDataLimitEnabled = true;
+      serverView.isDefaultDataLimitEnabled = true;
+    }
+  }
+
+  private async savePerKeyDataLimit(dialog: OutlinePerKeyDataLimitDialog) {
+    this.appRoot.showNotification(this.appRoot.localize('saving'));
+    const server = this.idServerMap.get(dialog.serverId());
+    const serverView = this.appRoot.getServerView(server.getId());
+    const keyId = dialog.keyId();
+    try {
+      if (dialog.dataLimitChanged()) {
+        const dataLimit = dialog.inputDataLimit();
+        await server.setAccessKeyDataLimit(keyId, displayDataAmountToDataLimit(dataLimit));
+        this.refreshTransferStats(server, serverView);
+      }
+      dialog.close();
+      this.appRoot.showNotification(this.appRoot.localize('saved'));
+    } catch (error) {
+      console.error(`Failed to set data limit for access key ${keyId}: ${error}`);
+      dialog.setInitialMenuState();
```

Comment:
Let's not change the state, so the user can see what was there.

Reply:
Done in be249e6